Core tests fail on Windows due to capitalisation issue with paths

### Discussed in https://github.com/beeware/toga/discussions/1566

<div type='discussions-op-text'>

<sup>Originally posted by **gregcowell** August 22, 2022</sup>
I have just setup my development environment. When I run the core test suite I am getting 8 failures.  I have omitted the successful test output below. The only thing I have done differently to the instructions is call my virtual environment directory "toga" instead of "venv".  I am using Python 3.10.5 on Windows 11.
```
(toga) PS C:\Users\gregc\python\toga\toga\src\core> python setup.py test

======================================================================
FAIL: test_as_file (tests.test_paths.TestPaths)
When started as `python app.py`, the app path is the folder holding app.py
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 90, in test_as_file
    self.assert_standalone_paths(output)
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 67, in assert_standalone_paths
    self.assertIn(
AssertionError: 'app.paths.toga=c:\\users\\gregc\\python\\toga\\toga\\src\\core\\toga' not found in ['app.paths.app=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\tests\\testbed', 'app.paths.data=C:\\Users\\gregc\\user_data\\org.testbed.standalone-app', 'app.paths.cache=C:\\Users\\gregc\\cache\\org.testbed.standalone-app', 'app.paths.logs=C:\\Users\\gregc\\logs\\org.testbed.standalone-app', 'app.paths.toga=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\toga']

======================================================================
FAIL: test_as_interactive (tests.test_paths.TestPaths)
At an interactive prompt, the app path is the current working directory
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 80, in test_as_interactive
    self.assert_standalone_paths(output)
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 67, in assert_standalone_paths
    self.assertIn(
AssertionError: 'app.paths.toga=c:\\users\\gregc\\python\\toga\\toga\\src\\core\\toga' not found in ['app.paths.app=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\tests\\testbed', 'app.paths.data=C:\\Users\\gregc\\user_data\\org.testbed.standalone-app', 'app.paths.cache=C:\\Users\\gregc\\cache\\org.testbed.standalone-app', 'app.paths.logs=C:\\Users\\gregc\\logs\\org.testbed.standalone-app', 'app.paths.toga=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\toga']

======================================================================
FAIL: test_as_module (tests.test_paths.TestPaths)
When started as `python -m app`, the app path is the folder holding app.py
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 100, in test_as_module
    self.assert_standalone_paths(output)
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 67, in assert_standalone_paths
    self.assertIn(
AssertionError: 'app.paths.toga=c:\\users\\gregc\\python\\toga\\toga\\src\\core\\toga' not found in ['app.paths.app=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\tests\\testbed', 'app.paths.data=C:\\Users\\gregc\\user_data\\org.testbed.standalone-app', 'app.paths.cache=C:\\Users\\gregc\\cache\\org.testbed.standalone-app', 'app.paths.logs=C:\\Users\\gregc\\logs\\org.testbed.standalone-app', 'app.paths.toga=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\toga']

======================================================================
FAIL: test_installed_as_module (tests.test_paths.TestPaths)
When the installed app is started, the app path is the folder holding app.py
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 193, in test_installed_as_module
    self.assertIn(f"app.paths.toga={Path(toga.__file__).parent}", results)
AssertionError: 'app.paths.toga=c:\\users\\gregc\\python\\toga\\toga\\src\\core\\toga' not found in ['app.paths.app=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\tests\\testbed\\installed', 'app.paths.data=C:\\Users\\gregc\\user_data\\org.testbed.installed', 'app.paths.cache=C:\\Users\\gregc\\cache\\org.testbed.installed', 'app.paths.logs=C:\\Users\\gregc\\logs\\org.testbed.installed', 'app.paths.toga=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\toga']

======================================================================
FAIL: test_simple_as_deep_file (tests.test_paths.TestPaths)
When a simple app is started as `python simple/app.py`, the app path is the folder holding app.py
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 155, in test_simple_as_deep_file
    self.assert_simple_paths(output)
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 120, in assert_simple_paths
    self.assertIn(
AssertionError: 'app.paths.toga=c:\\users\\gregc\\python\\toga\\toga\\src\\core\\toga' not found in ['app.paths.app=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\tests\\testbed\\simple', 'app.paths.data=C:\\Users\\gregc\\user_data\\org.testbed.simple-app', 'app.paths.cache=C:\\Users\\gregc\\cache\\org.testbed.simple-app', 'app.paths.logs=C:\\Users\\gregc\\logs\\org.testbed.simple-app', 'app.paths.toga=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\toga']

======================================================================
FAIL: test_simple_as_deep_module (tests.test_paths.TestPaths)
When a simple app is started as `python -m simple`, the app path is the folder holding app.py
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 165, in test_simple_as_deep_module
    self.assert_simple_paths(output)
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 120, in assert_simple_paths
    self.assertIn(
AssertionError: 'app.paths.toga=c:\\users\\gregc\\python\\toga\\toga\\src\\core\\toga' not found in ['app.paths.app=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\tests\\testbed\\simple', 'app.paths.data=C:\\Users\\gregc\\user_data\\org.testbed.simple-app', 'app.paths.cache=C:\\Users\\gregc\\cache\\org.testbed.simple-app', 'app.paths.logs=C:\\Users\\gregc\\logs\\org.testbed.simple-app', 'app.paths.toga=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\toga']

======================================================================
FAIL: test_simple_as_file_in_module (tests.test_paths.TestPaths)
When a simple app is started as `python app.py` inside a runnable module,
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 134, in test_simple_as_file_in_module
    self.assert_simple_paths(output)
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 120, in assert_simple_paths
    self.assertIn(
AssertionError: 'app.paths.toga=c:\\users\\gregc\\python\\toga\\toga\\src\\core\\toga' not found in ['app.paths.app=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\tests\\testbed\\simple', 'app.paths.data=C:\\Users\\gregc\\user_data\\org.testbed.simple-app', 'app.paths.cache=C:\\Users\\gregc\\cache\\org.testbed.simple-app', 'app.paths.logs=C:\\Users\\gregc\\logs\\org.testbed.simple-app', 'app.paths.toga=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\toga']

======================================================================
FAIL: test_simple_as_module (tests.test_paths.TestPaths)
When a simple apps is started as `python -m app` inside a runnable module,
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 145, in test_simple_as_module
    self.assert_simple_paths(output)
  File "c:\users\gregc\python\toga\toga\src\core\tests\test_paths.py", line 120, in assert_simple_paths
    self.assertIn(
AssertionError: 'app.paths.toga=c:\\users\\gregc\\python\\toga\\toga\\src\\core\\toga' not found in ['app.paths.app=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\tests\\testbed\\simple', 'app.paths.data=C:\\Users\\gregc\\user_data\\org.testbed.simple-app', 'app.paths.cache=C:\\Users\\gregc\\cache\\org.testbed.simple-app', 'app.paths.logs=C:\\Users\\gregc\\logs\\org.testbed.simple-app', 'app.paths.toga=C:\\Users\\gregc\\python\\toga\\toga\\src\\core\\toga']

----------------------------------------------------------------------
Ran 503 tests in 1.815s

FAILED (failures=8, skipped=1)
Test failed: <unittest.runner.TextTestResult run=503 errors=0 failures=8>
error: Test failed: <unittest.runner.TextTestResult run=503 errors=0 failures=8>
```</div>

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -109,6 +109,10 @@ jobs:
       run: |
         cd src/winforms
         pytest
+    - name: Core Test
+      run: |
+        cd src/core
+        pytest
 
   macOS:
     name: macOS backend tests

--- a/src/core/tests/test_paths.py
+++ b/src/core/tests/test_paths.py
@@ -26,46 +26,46 @@ class TestPaths(unittest.TestCase):
 
         self.assertEqual(
             app.paths.app,
-            Path.cwd(),
+            Path.cwd().resolve(),
         )
         self.assertEqual(
             app.paths.data,
-            Path.home() / "user_data" / "org.beeware.test-app",
+            Path.home().resolve() / "user_data" / "org.beeware.test-app",
         )
         self.assertEqual(
             app.paths.cache,
-            Path.home() / "cache" / "org.beeware.test-app",
+            Path.home().resolve() / "cache" / "org.beeware.test-app",
         )
         self.assertEqual(
             app.paths.logs,
-            Path.home() / "logs" / "org.beeware.test-app",
+            Path.home().resolve() / "logs" / "org.beeware.test-app",
         )
         self.assertEqual(
             app.paths.toga,
-            Path(toga.__file__).parent,
+            Path(toga.__file__).parent.resolve(),
         )
 
     def assert_standalone_paths(self, output):
         "Assert the paths for the standalone app are consistent"
         results = output.splitlines()
         self.assertIn(
-            f"app.paths.app={Path.cwd() / 'tests' / 'testbed'}",
+            f"app.paths.app={Path.cwd().resolve() / 'tests' / 'testbed'}",
             results,
         )
         self.assertIn(
-            f"app.paths.data={Path.home() / 'user_data' / 'org.testbed.standalone-app'}",
+            f"app.paths.data={Path.home().resolve() / 'user_data' / 'org.testbed.standalone-app'}",
             results,
         )
         self.assertIn(
-            f"app.paths.cache={Path.home() / 'cache' / 'org.testbed.standalone-app'}",
+            f"app.paths.cache={Path.home().resolve() / 'cache' / 'org.testbed.standalone-app'}",
             results,
         )
         self.assertIn(
-            f"app.paths.logs={Path.home() / 'logs' / 'org.testbed.standalone-app'}",
+            f"app.paths.logs={Path.home().resolve() / 'logs' / 'org.testbed.standalone-app'}",
             results,
         )
         self.assertIn(
-            f"app.paths.toga={Path(toga.__file__).parent}",
+            f"app.paths.toga={Path(toga.__file__).parent.resolve()}",
             results,
         )
 
@@ -103,22 +103,22 @@ class TestPaths(unittest.TestCase):
         "Assert the paths for the simple app are consistent"
         results = output.splitlines()
         self.assertIn(
-            f"app.paths.app={Path.cwd() / 'tests' / 'testbed' / 'simple'}",
+            f"app.paths.app={Path.cwd().resolve() / 'tests' / 'testbed' / 'simple'}",
             results,
         )
         self.assertIn(
-            f"app.paths.data={Path.home() / 'user_data' / 'org.testbed.simple-app'}",
+            f"app.paths.data={Path.home().resolve() / 'user_data' / 'org.testbed.simple-app'}",
             results,
         )
         self.assertIn(
-            f"app.paths.cache={Path.home() / 'cache' / 'org.testbed.simple-app'}",
+            f"app.paths.cache={Path.home().resolve() / 'cache' / 'org.testbed.simple-app'}",
             results,
         )
         self.assertIn(
-            f"app.paths.logs={Path.home() / 'logs' / 'org.testbed.simple-app'}", results
+            f"app.paths.logs={Path.home().resolve() / 'logs' / 'org.testbed.simple-app'}", results
         )
         self.assertIn(
-            f"app.paths.toga={Path(toga.__file__).parent}",
+            f"app.paths.toga={Path(toga.__file__).parent.resolve()}",
             results,
         )
 
@@ -175,19 +175,19 @@ class TestPaths(unittest.TestCase):
 
         results = output.splitlines()
         self.assertIn(
-            f"app.paths.app={Path.cwd() / 'tests' / 'testbed' / 'installed'}",
+            f"app.paths.app={Path.cwd().resolve() / 'tests' / 'testbed' / 'installed'}",
             results,
         )
         self.assertIn(
-            f"app.paths.data={Path.home() / 'user_data' / 'org.testbed.installed'}",
+            f"app.paths.data={Path.home().resolve() / 'user_data' / 'org.testbed.installed'}",
             results,
         )
         self.assertIn(
-            f"app.paths.cache={Path.home() / 'cache' / 'org.testbed.installed'}",
+            f"app.paths.cache={Path.home().resolve() / 'cache' / 'org.testbed.installed'}",
             results,
         )
         self.assertIn(
-            f"app.paths.logs={Path.home() / 'logs' / 'org.testbed.installed'}",
+            f"app.paths.logs={Path.home().resolve() / 'logs' / 'org.testbed.installed'}",
             results,
         )
-        self.assertIn(f"app.paths.toga={Path(toga.__file__).parent}", results)
+        self.assertIn(f"app.paths.toga={Path(toga.__file__).parent.resolve()}", results)

--- a/src/core/tests/test_paths.py
+++ b/src/core/tests/test_paths.py
@@ -26,42 +26,42 @@ class TestPaths(unittest.TestCase):
 
         self.assertEqual(
             app.paths.app,
-            Path.cwd().resolve(),
+            Path.cwd(),
         )
         self.assertEqual(
             app.paths.data,
-            Path.home().resolve() / "user_data" / "org.beeware.test-app",
+            Path.home() / "user_data" / "org.beeware.test-app",
         )
         self.assertEqual(
             app.paths.cache,
-            Path.home().resolve() / "cache" / "org.beeware.test-app",
+            Path.home() / "cache" / "org.beeware.test-app",
         )
         self.assertEqual(
             app.paths.logs,
-            Path.home().resolve() / "logs" / "org.beeware.test-app",
+            Path.home() / "logs" / "org.beeware.test-app",
         )
         self.assertEqual(
             app.paths.toga,
-            Path(toga.__file__).parent.resolve(),
+            Path(toga.__file__).parent,
         )
 
     def assert_standalone_paths(self, output):
         "Assert the paths for the standalone app are consistent"
         results = output.splitlines()
         self.assertIn(
-            f"app.paths.app={Path.cwd().resolve() / 'tests' / 'testbed'}",
+            f"app.paths.app={(Path.cwd() / 'tests' / 'testbed').resolve()}",
             results,
         )
         self.assertIn(
-            f"app.paths.data={Path.home().resolve() / 'user_data' / 'org.testbed.standalone-app'}",
+            f"app.paths.data={(Path.home() / 'user_data' / 'org.testbed.standalone-app').resolve()}",
             results,
         )
         self.assertIn(
-            f"app.paths.cache={Path.home().resolve() / 'cache' / 'org.testbed.standalone-app'}",
+            f"app.paths.cache={(Path.home() / 'cache' / 'org.testbed.standalone-app').resolve()}",
             results,
         )
         self.assertIn(
-            f"app.paths.logs={Path.home().resolve() / 'logs' / 'org.testbed.standalone-app'}",
+            f"app.paths.logs={(Path.home() / 'logs' / 'org.testbed.standalone-app').resolve()}",
             results,
         )
         self.assertIn(
@@ -103,19 +103,19 @@ class TestPaths(unittest.TestCase):
         "Assert the paths for the simple app are consistent"
         results = output.splitlines()
         self.assertIn(
-            f"app.paths.app={Path.cwd().resolve() / 'tests' / 'testbed' / 'simple'}",
+            f"app.paths.app={(Path.cwd() / 'tests' / 'testbed' / 'simple').resolve()}",
             results,
         )
         self.assertIn(
-            f"app.paths.data={Path.home().resolve() / 'user_data' / 'org.testbed.simple-app'}",
+            f"app.paths.data={(Path.home() / 'user_data' / 'org.testbed.simple-app').resolve()}",
             results,
         )
         self.assertIn(
-            f"app.paths.cache={Path.home().resolve() / 'cache' / 'org.testbed.simple-app'}",
+            f"app.paths.cache={(Path.home() / 'cache' / 'org.testbed.simple-app').resolve()}",
             results,
         )
         self.assertIn(
-            f"app.paths.logs={Path.home().resolve() / 'logs' / 'org.testbed.simple-app'}", results
+            f"app.paths.logs={(Path.home() / 'logs' / 'org.testbed.simple-app').resolve()}", results
         )
         self.assertIn(
             f"app.paths.toga={Path(toga.__file__).parent.resolve()}",
@@ -175,19 +175,19 @@ class TestPaths(unittest.TestCase):
 
         results = output.splitlines()
         self.assertIn(
-            f"app.paths.app={Path.cwd().resolve() / 'tests' / 'testbed' / 'installed'}",
+            f"app.paths.app={(Path.cwd() / 'tests' / 'testbed' / 'installed').resolve()}",
             results,
         )
         self.assertIn(
-            f"app.paths.data={Path.home().resolve() / 'user_data' / 'org.testbed.installed'}",
+            f"app.paths.data={(Path.home() / 'user_data' / 'org.testbed.installed').resolve()}",
             results,
         )
         self.assertIn(
-            f"app.paths.cache={Path.home().resolve() / 'cache' / 'org.testbed.installed'}",
+            f"app.paths.cache={(Path.home() / 'cache' / 'org.testbed.installed').resolve()}",
             results,
         )
         self.assertIn(
-            f"app.paths.logs={Path.home().resolve() / 'logs' / 'org.testbed.installed'}",
+            f"app.paths.logs={(Path.home() / 'logs' / 'org.testbed.installed').resolve()}",
             results,
         )
         self.assertIn(f"app.paths.toga={Path(toga.__file__).parent.resolve()}", results)

